packages/ai/src/LLMStepInstance.ts

```diff
@@ -386,21 +425,23 @@ export function serializeStepParams(
         visitor.artifact(input),
       ])
     ),
-    outputIds: Object.fromEntries(
-      Object.entries(params.outputs)
-        .map(([key, output]) =>
-          output ? [key, visitor.artifact(output)] : undefined
-        )
-        .filter((x) => x !== undefined)
-    ),
   };
 }
 
+type SerializedState<Shape extends IOShape> =
+  | (Omit<Extract<StepState<Shape>, { kind: "DONE" }>, "outputs"> & {
+      outputIds: Record<string, number>;
+    })
+  | Exclude<StepState<Shape>, { kind: "DONE" }>;
+
 export type SerializedStep = Omit<
   StepParams<IOShape>,
-  "inputs" | "outputs" | "template"
+  "inputs" | "outputs" | "template" | "state"
 > & {
   templateName: string;
   inputIds: Record<string, number>;
-  outputIds: Record<string, number>;
+  // Legacy - in the modern format we store outputs on SerializedState, but we still support this field for old workflows.
+  // Can be removed if we deserialize and serialize again all workflows in the existing database.
+  outputIds?: Record<string, number>;
```

Comment:
I hope I haven't broken old workflows with this change (moving outputs to `step.state`, and changing serialization code accordingly); as far as I can tell it's fine, i.e. old workflows still load in my dev hub instance.